src/compiler/checker.ts

```diff
@@ -4415,6 +4415,21 @@ namespace ts {
 
                 return Debug.fail("Should be unreachable.");
 
+
+                function typeToTypeNodeOrCircularityEllision(type: Type) {
+                    if (type.flags & TypeFlags.Union) {
+                        if (context.visitedTypes && context.visitedTypes.has("" + getTypeId(type))) {
+                            if (!(context.flags & NodeBuilderFlags.AllowAnonymousIdentifier)) {
+                                context.encounteredError = true;
+                                context.tracker?.reportCyclicStructureError?.();

```

Comment:
Since you fixed the failure case separately, can you think of a way to trigger this? Having a test would be nice.

Reply:
I'd have to create an instance where the type doesn't adopt an alias symbol.... Maybe doable by enclosing the indexed access in a union? I'll see if I can add a test.